optuna/cli.py

```diff
@@ -50,6 +48,174 @@ def _check_storage_url(storage_url: Optional[str]) -> str:
     return storage_url
 
 
+def _format_value(value: Any) -> Any:
+    #  Format value that can be serialized to JSON or YAML
+    if value is None or isinstance(value, (int, float)):
+        return value
+    elif isinstance(value, datetime.datetime):
+        return value.strftime(_DATETIME_FORMAT)
+    elif isinstance(value, list):
+        return list(_format_value(v) for v in value)
+    elif isinstance(value, tuple):
+        return tuple(_format_value(v) for v in value)
+    elif isinstance(value, dict):
+        return {_format_value(k): _format_value(v) for k, v in value.items()}
+    else:
+        return str(value)
+
+
+def _convert_to_dict(
+    records: List[Dict[Tuple[str, str], Any]], columns: List[Tuple[str, str]], flatten: bool
+) -> Tuple[List[Dict[str, Any]], List[str]]:
+    header = []
+    ret = []
+    if flatten:
+        for column in columns:
+            if column[1] != "":
+                header.append(f"{column[0]}_{column[1]}")
+            elif any(isinstance(record.get(column), (list, tuple)) for record in records):
+                max_length = 0
+                for record in records:
+                    if column in record:
+                        max_length = max(max_length, len(record[column]))
+                for i in range(max_length):
+                    header.append(f"{column[0]}_{i}")
+            else:
+                header.append(column[0])
+        for record in records:
+            row = {}
+            for column in columns:
+                if column in record:
+                    value = _format_value(record[column])
+                    if column[1] != "":
+                        row[f"{column[0]}_{column[1]}"] = value
+                    elif any(isinstance(record.get(column), (list, tuple)) for record in records):
+                        for i, v in enumerate(value):
+                            row[f"{column[0]}_{i}"] = v
+                    else:
+                        row[f"{column[0]}"] = value
```

Comment:
How about using the early-return pattern to reduce the nest level?
```suggestion
                if column not in record:
                    continue
                value = _format_value(record[column])
                if column[1] != "":
                    row[f"{column[0]}_{column[1]}"] = value
                elif any(isinstance(record.get(column), (list, tuple)) for record in records):
                    for i, v in enumerate(value):
                        row[f"{column[0]}_{i}"] = v
                else:
                    row[f"{column[0]}"] = value
```